src/nvim/terminal.c

```diff
@@ -0,0 +1,997 @@
+// VT220/xterm-like terminal emulator implementation for Neovim. Powered by
+// libvterm(http://www.leonerd.org.uk/code/libvterm/).
+//
+// libvterm is a pure C99 terminal emulation library with abstract input and
+// display. This means that the library needs to read data from the master fd
+// and feed VTerm instances, which will invoke user callbacks with screen
+// update instructions that must be mirrored to the real display.
+//
+// Keys are pressed in VTerm instances by calling
+// vterm_keyboard_key/vterm_keyboard_unichar, which generates byte streams that
+// must be fed back to the master fd.
+//
+// This implementation uses Neovim buffers as the display mechanism for both the
+// visible screen and the scrollback buffer. When focused, the window "pins" to
+// the bottom of the buffer and mirrors libvterm screen state.
+//
+// When a line becomes invisible due to a decrease in screen height or because a
+// line was pushed up during normal terminal output, we store the line
+// information in the scrollback buffer which is mirrored in the Neovim buffer
```

Comment:
Perhaps, but I suggested the parens instead because the sentence is already long and hard to follow. Both should work.